### ISSUE TITLE
Implementation

### ISSUE THREAD

**Issue by jeanpierrethach** [jeanpierrethach/Neural-Style-Transfer#2]
# Concepts to understand

|                    [Machine Learning Coursera's](https://www.coursera.org/learn/machine-learning) by Andrew Ng                   | 
|:------------------------------------------------------------:|
|                      Mean squared error                      |
|                   Normalization, Rescaling                   |
|                   Cost function                   |
|          Activation functions (tanh, sigmoid, etc.)          |
|                       Gradient descent                       |
|                        Backpropagation                       |
| Optimizer function  (overfitting, regularization/penalizing) |

| [Gatys Image Style Transfer paper](http://www.cv-foundation.org/openaccess/content_cvpr_2016/papers/Gatys_Image_Style_Transfer_CVPR_2016_paper.pdf)  |
|:--------------------------------:|
|       Content loss function      |
|        Style loss function       |
|       [Gram Matrix](https://en.wikipedia.org/wiki/Gramian_matrix)       |

* Denoising loss function

* Convolutional Neural Network
   1. Filtering
   2. Convolution layer
   3. Pooling layer
   4. Rectified Linear Units layer
   5. Fully connected layer

## Papers
_Image Style Transfer Using Convolutional Neural Networks_
* http://www.cv-foundation.org/openaccess/content_cvpr_2016/papers/Gatys_Image_Style_Transfer_CVPR_2016_paper.pdf

_Preserving Color in Neural Artistic Style Transfer_
* https://arxiv.org/pdf/1606.05897.pdf

## Courses
_Coursera's Machine Learning by Andrew Ng_
* https://www.coursera.org/learn/machine-learning 

_CS231n Convolutional Neural Networks for Visual Recognition_
* http://cs231n.github.io/ 
* [CS231n Winter 2016 Stanford Playlist](https://www.youtube.com/playlist?list=PLLvH2FwAQhnpj1WEB-jHmPuUeQ8mX-XXG)

_Coursera's Neural Networks by Geoffrey Hinton_
* https://www.coursera.org/learn/neural-networks 

## Other resources
_Deep Learning by Ian Goodfellow, Yoshua Bengio and Aaron Courville_
* http://www.deeplearningbook.org/

_MOOCs, books, etc._
* https://www.reddit.com/r/MachineLearning/wiki/index

_Tensorflow resources_
* https://github.com/jtoy/awesome-tensorflow
